controllers/cloudstackisolatednetwork_controller.go

```diff
@@ -0,0 +1,102 @@
+	"context"
+
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	infrav1 "github.com/aws/cluster-api-provider-cloudstack/api/v1beta1"
+	csCtrlrUtils "github.com/aws/cluster-api-provider-cloudstack/controllers/utils"
+	"github.com/aws/cluster-api-provider-cloudstack/pkg/cloud"
+)
+
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=cloudstackisolatednetworks,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=cloudstackisolatednetworks/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=cloudstackisolatednetworks/finalizers,verbs=update
+
+// CloudStackIsoNetReconciler reconciles a CloudStackZone object
+type CloudStackIsoNetReconciler struct {
+	csCtrlrUtils.ReconcilerBase
+}
+
+// CloudStackZoneReconciliationRunner is a ReconciliationRunner with extensions specific to CloudStackCluster reconciliation.
+type CloudStackIsoNetReconciliationRunner struct {
+	csCtrlrUtils.ReconciliationRunner
+	Zone                  *infrav1.CloudStackZone
+	ReconciliationSubject *infrav1.CloudStackIsolatedNetwork
+	CSUser                cloud.Client
+}
+
+// Initialize a new CloudStackIsoNet reconciliation runner with concrete types and initialized member fields.
+func NewCSIsoNetReconciliationRunner() *CloudStackIsoNetReconciliationRunner {
+	// Set concrete type and init pointers.
+	r := &CloudStackIsoNetReconciliationRunner{ReconciliationSubject: &infrav1.CloudStackIsolatedNetwork{}}
+	r.Zone = &infrav1.CloudStackZone{}
+	// Setup the base runner. Initializes pointers and links reconciliation methods.
+	r.ReconciliationRunner = csCtrlrUtils.NewRunner(r, r.ReconciliationSubject)
+	return r
+}
+
+func (reconciler *CloudStackIsoNetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, retErr error) {
+	return NewCSIsoNetReconciliationRunner().
+		UsingBaseReconciler(reconciler.ReconcilerBase).
+		ForRequest(req).
+		WithRequestCtx(ctx).
+		RunBaseReconciliationStages()
+}
+
+func (r *CloudStackIsoNetReconciliationRunner) Reconcile() (retRes ctrl.Result, retErr error) {
+	if res, err := r.RequeueIfMissingBaseCRDs(); r.ShouldReturn(res, err) {
+		return res, err
+	}
+	if res, err := r.GetParent(r.ReconciliationSubject, r.Zone)(); r.ShouldReturn(res, err) {
+		return res, err
+	}
+	controllerutil.AddFinalizer(r.ReconciliationSubject, infrav1.IsolatedNetworkFinalizer)
+
+	// Setup isolated network, endpoint, egress, and load balancing.
+	// Set endpoint of CloudStackCluster if it is not currently set. (uses patcher to do so)
+	if csClusterPatcher, err := patch.NewHelper(r.CSCluster, r.Client); err != nil {
+		return r.ReturnWrappedError(retErr, "error encountered while setting up CloudStackCluster patcher")
+	} else if err := r.CS.GetOrCreateIsolatedNetwork(r.Zone, r.ReconciliationSubject, r.CSCluster); err != nil {
```

Comment:
Why the chained else-ifs?  I think they obfuscate unnecessarily and add no value in this case.

Reply:
I can break this all out. 

It will just be a bit more verbose. The else if maintains one scope across the entire conditional block.

Using successive ifs will require declaring vars prior to the if statement.

I'll review. I might even just run them as stages.